packages/components/src/color-picker/inputs.js

```diff
@@ -18,24 +18,18 @@ import IconButton from '../icon-button';
 import { isValidHex } from './utils';
 import TextControl from '../text-control';
 
+const toLowerCase = ( value ) => String( value ).toLowerCase();
+
 /* Wrapper for TextControl, only used to handle intermediate state while typing. */
 class Input extends Component {
 	constructor( { value } ) {
 		super( ...arguments );
-		this.state = { value: String( value ).toLowerCase() };
+		this.state = { value: toLowerCase( value ) };
 		this.handleBlur = this.handleBlur.bind( this );
 		this.handleChange = this.handleChange.bind( this );
 		this.handleKeyDown = this.handleKeyDown.bind( this );
 	}
 
-	componentWillReceiveProps( nextProps ) {
```

Comment:
I think removing componentWillReceiveProps,  has a small behavior change. Previously if the value was changed externally after the constructor is executed ( e.g: a plugin changing the block attribute value), to contain uppercase letters we would make the letters lowercase. Now the letters continue uppercase.
I think to solve this we would need getDerivedState from props. I'm not certain if it is worth it as it would have a performance impact and this approach seems fine.

Reply:
I have no idea if I'm right anymore :) This logic is quite complex and hard to follow. There are some cases where you trigger change outside and try to mirror local state and in other places it only updates local state 🤷‍♂️

Reply:
Just to note that I'm investigating this and trying to come up with some tests that guarantee behavior works as expected.

Reply:
Tests added so it's now clear what fails on removing `componentWillReceiveProps` method and what happens when it's substituted by `getDerivedStateFromProps`.

Will ping you for a review when all tests pass. I hadn't done that yet because I was still working on it. Marking it as "In Progress" for clarification.